auth/authentication.go

```diff
@@ -5,41 +5,62 @@ import (
 	"net/http"
 
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/dp-search-api/models"
 )
 
 // Authenticator structure which holds the secret key for validating clients. This will be replaced in the future, after the `thin-slices` has been delivered
 type Authenticator struct {
-	SecretKey  string
-	HeaderName string
+	SecretKey  				string
+	HeaderName 				string
+	HasPrivateEndpoints     bool
```

Comment:
As stated with the GET endpoint, any authentication should take place based on the correct auth tokens being set and should not be tied to whether the application is in web or publishing - these two things are not the same thing.